### DISCUSSION TITLE
How to scale ui of the program?

### DISCUSSION THREAD

**Discussion by ghost** [mapeditor/tiled#3874] [Q&A] [answered]
How to scale ui of the program? It is too tiny...

**Comment by bjorn** [comment]
There's likely some environment variables you can use to help with this, see https://doc.qt.io/qt-6/highdpi.html#environment-variable-reference.

However, please do provide more details about your OS and maybe some screenshots. Maybe it is a problem that can be addressed without needing workarounds.

**Reply by ghost** [reply]
I use an old 1.8 version from debian12 package. IIUC it uses qt5 which also has `QT_SCALE_FACTOR` but it doesn't affect

System: Debian12 + sway (wayland) + (enabled Xwayland if it matters)

**Reply by sonic2kk** [reply]
Xwayland can affect the scaling of applications, particularly if you're using a scaled display on Wayland.

If you can, try using native Wayland. If you still have issues, try setting `QT_AUTO_SCREEN_SCALE_FACTOR=1`.

**Reply by bjorn** [reply]
You may also want to try setting `QT_FONT_DPI` to a higher value. Generally Tiled does not need scaling and will adjust appropriately to the DPI (also scaling up margins in the UI and line widths in some places).

**Reply by ghost** [reply]
I just tried `QT_FONT_DPI`. No effect. I am almost sure I need scaling not in the sense of DPI auto adjusting but in the sense of accessibility feature

**Reply by bjorn** [reply]
Hmm, I've never seen `QT_FONT_DPI` not having any effect... What value did you try setting it to? Something like `QT_FONT_DPI=192` should make for quite large fonts / UI.

**Reply by ghost** [reply]
Honestly, I tried `QT_FONT_DPI=2` but 192 also don't have effect...

![screen-1705229469](https://github.com/mapeditor/tiled/assets/97330399/22452e8e-234d-4ec9-a72f-b5ef23cda6a0)

![screen-1705229497](https://github.com/mapeditor/tiled/assets/97330399/866145aa-9594-4e24-90fc-cc64f3f64286)

**Reply by bjorn** [reply, marked as answer]
@dustdfg You can set an environment variable for a child process using either:
```
$ export QT_FONT_DPI=196
$ tiled
```
Or:
```
$ QT_FONT_DPI=196 tiled
```
Otherwise, you're only changing the value for the terminal itself.

**Comment by ghost** [comment]
> Xwayland can affect the scaling of applications, particularly if you're using a scaled display on Wayland.
> 
> If you can, try using native Wayland. If you still have issues, try setting `QT_AUTO_SCREEN_SCALE_FACTOR=1`.

I just tried to disable Xwayland. The result is same. Using `QT_AUTO_SCREEN_SCALE_FACTOR=1` with or without enabled Xwayland didn't have an impact

About wayland display scaling. Sway distributed as debain12 package doesn't support fractional scaling so I used any other workarounds like increase font size of the bar and etc. **but scaling factor of the display is 1**

**Comment by ghost** [comment]
> QT_FONT_DPI=196 tiled

Yeah, I am a fool. I tried this form too but I added `;` between out of habit. Thank you @bjorn
